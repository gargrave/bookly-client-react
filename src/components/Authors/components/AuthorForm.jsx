import React from 'react'
import { func, shape, string } from 'prop-types'

const AuthorForm = props => {
  return (
    <form noValidate style={{ maxWidth: '500px', margin: 'auto', marginTop: '20px', textAlign: 'left' }}>
      <div className="input-single">
        <label htmlFor="firstName">First name:</label>
        <input
          type="text"
          name="firstName"
          id="firstName"
          value={props.author.firstName}
          onChange={props.handleInputChange}
        />
      </div>

      <div className="input-single">
        <label htmlFor="lastName">Last name:</label>
        <input
          type="text"
          name="lastName"
          id="lastName"
          value={props.author.lastName}
          onChange={props.handleInputChange}
        />
      </div>

      <div className="input-single">
        <button type="button" onClick={props.handleSubmit}>
          Submit
        </button>
        <button type="button" className="button-info float-right" onClick={props.handleCancel}>
          Cancel
        </button>
      </div>
    </form>
  )
}

AuthorForm.propTypes = {
  author: shape({
    firstName: string.isRequired,
    lastName: string.isRequired
  }).isRequired,
  handleInputChange: func.isRequired,
  handleSubmit: func.isRequired,
  handleCancel: func.isRequired
}

export default AuthorForm
