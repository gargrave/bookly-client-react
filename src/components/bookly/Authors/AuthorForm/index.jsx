import React from 'react'
import { func, shape, string } from 'prop-types'

import { buildClasses } from '@/utils/cssHelpers'

import '@/components/common/Form/styles.css'

const AuthorForm = ({
  author,
  onCancel,
  onInputChange,
  onSubmit,
}) => {
  return (
    <form
      className={buildClasses(['form', 'author-form'])}
      style={{ maxWidth: '500px', margin: 'auto', marginTop: '20px', textAlign: 'left' }}
      onSubmit={onSubmit}
      noValidate>
      <div className="input-single">
        <label htmlFor="firstName">First name:</label>
        <input
          type="text"
          name="firstName"
          id="firstName"
          value={author.firstName}
          onChange={onInputChange}
        />
      </div>

      <div className="input-single">
        <label htmlFor="lastName">Last name:</label>
        <input
          type="text"
          name="lastName"
          id="lastName"
          value={author.lastName}
          onChange={onInputChange}
        />
      </div>

      <div className="input-single">
        <button type="submit" className="submit-button" onClick={onSubmit}>
          Submit
        </button>
        <button type="button" className="button-info float-right cancel-button" onClick={onCancel}>
          Cancel
        </button>
      </div>
    </form>
  )
}

AuthorForm.propTypes = {
  author: shape({
    firstName: string.isRequired,
    lastName: string.isRequired,
  }).isRequired,
  onCancel: func.isRequired,
  onInputChange: func.isRequired,
  onSubmit: func.isRequired,
}

export default AuthorForm
