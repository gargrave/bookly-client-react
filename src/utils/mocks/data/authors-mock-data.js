export const authorsMockData = [
  {
    id: 8,
    firstName: 'Aldous',
    lastName: 'Huxley',
    createdAt: '2017-07-01T01:16:43.881Z',
    updatedAt: '2017-07-01T01:16:43.881Z'
  },
  {
    id: 51,
    firstName: 'Kurt',
    lastName: 'Vonnegut',
    createdAt: '2017-07-06T22:02:11.504Z',
    updatedAt: '2017-07-06T22:02:11.504Z'
  },
  {
    id: 59,
    firstName: 'George R. R.',
    lastName: 'Martin',
    createdAt: '2017-08-17T20:47:35.232Z',
    updatedAt: '2017-08-17T20:47:35.232Z'
  },
  {
    id: 52,
    firstName: 'Tom',
    lastName: 'Robbins',
    createdAt: '2017-07-06T22:02:17.051Z',
    updatedAt: '2017-08-17T21:27:35.885Z'
  }
]
