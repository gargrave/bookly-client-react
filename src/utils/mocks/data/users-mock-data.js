export const usersMockData = [
  {
    id: 1,
    email: 'mockuser1@gmail.com',
    createdAt: '2017-06-30T19:52:54.644Z',
    updatedAt: '2017-06-30T19:52:54.644Z',
    lastLogin: '2017-07-01T01:11:03.942Z',
    verified: true,
  },
];
