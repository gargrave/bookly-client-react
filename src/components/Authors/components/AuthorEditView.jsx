import React from 'react'
import { func, number, shape, string } from 'prop-types'

import AuthorForm from '../components/AuthorForm'

const AuthorDetailView = props => {
  return (
    <div>
      <h2>Edit Author</h2>
      <AuthorForm
        author={props.author}
        handleInputChange={props.handleInputChange}
        handleSubmit={props.handleSubmit}
        handleCancel={props.handleCancel}
      />
    </div>
  )
}

AuthorDetailView.propTypes = {
  author: shape({
    id: number,
    firstName: string,
    lastName: string,
    createdAt: string,
    updatedAt: string
  }),
  handleInputChange: func.isRequired,
  handleSubmit: func.isRequired,
  handleCancel: func.isRequired
}

export default AuthorDetailView
