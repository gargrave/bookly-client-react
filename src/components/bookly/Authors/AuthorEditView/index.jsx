// @flow
import React from 'react';
import { func, number, shape, string } from 'prop-types';

import type { Author } from '../../../../constants/flowtypes';

import AuthorForm from '../AuthorForm';

type Props = {
  author: Author,
  onCancel: Function,
  onInputChange: Function,
  onSubmit: Function,
};

function AuthorEditView({
  author,
  onCancel,
  onInputChange,
  onSubmit,
}: Props) {
  return (
    <div className="author-edit-view">
      <h2>Edit Author</h2>
      <AuthorForm
        author={author}
        onCancel={onCancel}
        onInputChange={onInputChange}
        onSubmit={onSubmit} />
    </div>
  );
}

AuthorEditView.propTypes = {
  author: shape({
    id: number,
    firstName: string,
    lastName: string,
    createdAt: string,
    updatedAt: string,
  }),
  onCancel: func.isRequired,
  onInputChange: func.isRequired,
  onSubmit: func.isRequired,
};

export default AuthorEditView;
