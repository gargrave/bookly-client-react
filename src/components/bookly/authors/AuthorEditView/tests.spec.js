import React from 'react';
import { shallow } from 'enzyme';

import { authorsMockData } from '../../../..//utils/mocks/data/authors-mock-data';

import AuthorEditView from './';
import AuthorForm from '../../authors/AuthorForm';

describe('AuthorEditView', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      author: Object.create(authorsMockData[0]),
      onCancel: jest.fn(),
      onInputChange: jest.fn(),
      onSubmit: jest.fn(),
    };

    component = shallow(<AuthorEditView {...props} />);
  });

  test('renders correctly', () => {
    const element = component.find('.author-edit-view');
    expect(component).toMatchSnapshot();
    expect(element.length).toEqual(1);
  });

  test('renders the AuthorForm component', () => {
    expect(component.find(AuthorForm).length).toEqual(1);
  });
});
