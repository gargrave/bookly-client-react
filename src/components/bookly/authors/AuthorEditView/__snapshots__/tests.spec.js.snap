// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorEditView renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthorDetailView
    author={Object {}}
    onCancel={[Function]}
    onInputChange={[Function]}
    onSubmit={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Edit Author
        </h2>,
        <AuthorForm
          author={Object {}}
          onCancel={[Function]}
          onInputChange={[Function]}
          onSubmit={[Function]}
        />,
      ],
      "className": "author-edit-view",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Edit Author",
        },
        "ref": null,
        "rendered": "Edit Author",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "author": Object {},
          "onCancel": [Function],
          "onInputChange": [Function],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Edit Author
          </h2>,
          <AuthorForm
            author={Object {}}
            onCancel={[Function]}
            onInputChange={[Function]}
            onSubmit={[Function]}
          />,
        ],
        "className": "author-edit-view",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Edit Author",
          },
          "ref": null,
          "rendered": "Edit Author",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "author": Object {},
            "onCancel": [Function],
            "onInputChange": [Function],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
